landlab/components/lateral_erosion/lateral_erosion.py

```diff
@@ -0,0 +1,631 @@
+        "dzlat": "Change in elevation at each node from lateral erosion during time step",
+        "vollat": "Array tracking volume eroded at each node from lateral erosion",
+        "qs_in": "Volume per unit time of sediment entering each node"
+    }
+
+    def __init__(
+        self,
+        grid,
+        latero_mech="UC",
+        alph=0.8,
+        Kv=None,
+        Kl_ratio=1.0,
+        solver="basic",
+        inlet_on=False,
+        inlet_node=None,
+        inlet_area=None,
+        qsinlet=0.0,
+    ):
+        if solver not in ("basic", "adaptive"):
+            raise ValueError(
+                "value for solver not understood ({val} not one of {valid})".format(
+                    val=solver, valid=", ".join(("basic", "adaptive")))
+            )
+
+        if latero_mech not in ("UC", "TB"):
+            raise ValueError(
+                "value for latero_mech not understood ({val} not one of {valid})".format(
+                    val=latero_mech, valid=", ".join(("UC", "TB")))
+            )
+
+        if inlet_on and (inlet_node is None or inlet_area is None):
+            raise ValueError(
+                "inlet_on is True, but no inlet_node or inlet_area is provided."
+            )
+
+        if Kv is None:
+            raise ValueError(
+                "Kv must be set as a float, node array, or field name. It was None."
+            )
+
+        self._grid = grid
+        # Create fields needed for this component if not already existing
+        if "volume__lateral_erosion" in grid.at_node:
+            self.vol_lat = grid.at_node["volume__lateral_erosion"]
+        else:
+            self.vol_lat = grid.add_zeros("volume__lateral_erosion", at="node")
+        if "qs_in" in grid.at_node:
+            self.qs_in = grid.at_node["qs_in"]
+        else:
+            self.qs_in = grid.add_zeros("qs_in", at="node")
+
+        # you can specify the type of lateral erosion model you want to use.
+        # But if you don't the default is the undercutting-slump model
+        if latero_mech == "TB":
+            self._TB = True
+            self._UC = False
+        else:
+            self._UC = True
+            self._TB = False
+        # option use adaptive time stepping. Default is fixed dt supplied by user
+        if solver == "basic":
+            self.run_one_step = self.run_one_step_basic
+        elif solver == "adaptive":
+            self.run_one_step = self.run_one_step_adaptive
+            self.frac = 0.3  # for time step calculations
+        self.alph = alph
+        self.Kv = Kv  # can be overwritten with spatially variable
+        self.inlet_on = False  # will be overwritten below if inlet area is provided
+        self.Klr = float(Kl_ratio)  # default ratio of Kv/Kl is 1. Can be overwritten
+        self.wid_coeff = 0.4    #coefficient for calculating channel width
+        self.wid_exp = 0.35    #exponent for calculating channel width
```

Comment:
Similar to the comment about the CFL component. I'd recommend stating that these are hard coded coefficients somewhere in the docs.

Reply:
Hard coded coefficients were moved to the top of the file.